subprojects/core/src/main/java/org/gradle/api/internal/file/collections/ImmutableFileCollection.java

```diff
@@ -72,6 +80,10 @@ public TaskDependency getBuildDependencies() {
             this.files = ImmutableSet.copyOf(files);
         }
 
+        FileOnlyImmutableFileCollection(Iterable<File> files) {
+            this.files = ImmutableSet.copyOf(files);
+        }
+
```

Comment:
Nitpick: we could have a single constructor here with `ImmutableSet<File>` as its parameter.